docs/guides/python/blender-render.mdx

```diff
@@ -0,0 +1,893 @@
+---
+description: Use the Nitric framework to build a service for rendering Blender scenes using cloud GPUs
+tags:
+  - API
+  - AI & Machine Learning
+languages:
+  - python
+---
+
+# Use Cloud GPUs for rendering your Blender projects
+
+This example shows how you can create a remote Blender rendering application using Blender's Python interface.
+
+By using the cloud you can render your Blender scenes on infrastructure that scales and with CPU or GPU resources you might not have access to locally.
+
+Here's a final render that was done using cloud GPUs:
+
+<div>
+  <div className="video-container">
+    <iframe
+      src="https://www.youtube-nocookie.com/embed/JoiAtMZ3HME"
+      title="Nitric Logo Blender"
+      frameBorder="0"
+      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+      allowFullScreen
+    />
+  </div>
+</div>
+
+## Prerequisites
+
+- [UV](https://docs.astral.sh/uv/) - for dependency management
+- The [Nitric CLI](/get-started/installation)
+- An [AWS](https://aws.amazon.com) or [Google Cloud](https://cloud.google.com) account (_your choice_)
+
+## Getting started
+
+We'll start by creating a new Nitric project.
+
+<Note>
+  If you want to take a look at the finished code, it can be found
+  [here](https://github.com/nitrictech/examples/tree/main/v1/blender-render).
+</Note>
+
+```bash
+nitric new blender-rendering py-starter
+```
+
+We can then resolve our dependencies.
+
+```bash
+uv sync
+```
+
+We'll also add [bpy](https://docs.blender.org/api/current/info_overview.html#python-in-blender) as an optional dependency. By making it an optional dependency, we can choose to only install it for the containers that require it, reducing the size of the containers. `bpy` is the API that will interact with the Blender environment for setting up and rendering our scenes. The version of `bpy` should match the version of Blender you intend on using.
+
+```bash
+uv add bpy==4.2.0 --optional ml
+```
+
+We'll organize our project structure like so:
+
+```text
++--common/
+|  +-- __init__.py
+|  +-- resources.py
++--batches/
+|  +-- renderer.py
++--services/
+|  +-- api.py
++-- blender.dockerfile
++-- blender.dockerignore
++--.gitignore
++--.python-version
++-- pyproject.toml
++-- python.dockerfile
++-- python.dockerignore
++-- nitric.yaml
++-- README.md
+```
+
+## Creating the resources
+
+We'll start by creating a file to define our Nitric resources. For this project we'll need an API, Batch Job, and two buckets, one for the `.blend` files and one for the resulting renders. The API will interface with the buckets, while the Batch Job will handle the long-running renders.
+
```

Comment:
```suggestion

```bash
mkdir common
touch common/resources.py
touch common/__init__.py
```

```

Reply:
Not sure if this cut off the end

Reply:
I wasn't going to show the creating of each file since its shown in the project structure